docs/zetacore/gas_and_fees.md

```diff
@@ -0,0 +1,334 @@
+
+* **Cosmos Transaction Fees**: As a Layer 1 blockchain built with Cosmos SDK, sending direct Cosmos transactions on ZetaChain involves using the Cosmos gas mechanism for fee calculation and payment.
+
+## ZetaChain EVM Transaction Fees
+
+When users interact with the ZetaChain EVM, for any operations, they will pay fees using the standard EVM fee mechanism.
+More information can be found at the [EVM Fees Documentation](https://ethereum.org/developers/docs/gas/).
+
+ZetaChain currently uses Go-Ethereum `v1.15.11` and supports both legacy and EIP-1559 transaction types.
+
+ZetaChain EVM currently doesn't support `EIP-7702` for sponsored transactions.
+
+The base fee parameter can be queried using the ETH JSON-RPC:
+
+```shell
+curl -X POST https://zetachain-evm.blockpi.network/v1/rpc/public \
+  -H "Content-Type: application/json" \
+  -d '{
+    "jsonrpc": "2.0",
+    "method": "eth_getBlockByNumber",
+    "params": ["latest", false],
+    "id": 1
+  }'
+```
+
+Or using the Cast CLI:
+
+```shell
+cast block latest --rpc-url https://zetachain-evm.blockpi.network/v1/rpc/public
+```
+
+## Connected Chains Withdraw Fees
+
+Users create outgoing cross-chain transactions to connected chains using the withdraw interface of the ZetaChain gateway:
+- `withdraw`
+- `withdrawAndCall`
+- `call`
+
+As functions called on the ZetaChain EVM, the standard EVM transaction fee mechanism applies for executing these functions on ZetaChain.
+
+In addition to the EVM transaction fees, users must also pay additional fees to cover the cost of executing the transaction on the connected chain.
+
+The additional withdraw fees can be queried on the ZRC20 asset contract with the following functions:
+```solidity
+// for simple withdraws, default gas limit set per zrc20
+function withdrawGasFee() external view returns (address, uint256);
+
+// for withdraws that include a call on the connected chain
+function withdrawGasFeeWithGasLimit(uint256 gasLimit) external view returns (address, uint256);
+```
+
+Users must approve the gateway contract to spend the required fee amount in addition to the withdraw amount using the ERC20 standard.
+
+For each connected chain, the ZetaChain observer set votes on and maintains an updated "gas price".
+The gas price and gas limit are two parameters used to calculate the withdraw fees on connected chains.
+We use this terminology for all chains for simplicity and consistency, although not all connected chains use the same concepts, as we support non-EVM chains.
+The meaning of gas price and gas limit depends on the connected chain.
+
+These differences are summarized below.
+
+### EVM
+
+For EVM-compatible chains, ZetaChain relies on the standard Ethereum gas mechanism.
+More information about Ethereum gas can be found in the [Ethereum Gas Documentation](https://ethereum.org/developers/docs/gas/).
+
+The withdraw fee for EVM chains is calculated using the gas price and gas limit parameters:
```

Comment:
Should we specify that this gas limit is set by the user, and different from the gas limit for the, zevm transaction creating the withdraw

Reply:
calloptions.Gaslimit is optional

Reply:
I think it will be ok for now, might revisit in the future